filer/models/filemodels.py

```diff
@@ -309,7 +309,10 @@ def save(self, *args, **kwargs):
             self.generate_sha1()
         except (IOError, TypeError, ValueError) as e:
             pass
-        if filer_settings.FOLDER_AFFECTS_URL and (self._is_path_changed() or (self._old_sha1 != self.sha1)):
+        replaced_file = self._old_sha1 != self.sha1
+        if filer_settings.FOLDER_AFFECTS_URL and (self._is_path_changed() or replaced_file):
+            if replaced_file:
+                self.name = None  # if new file submitted for same id we overwrite what was previously in name
```

Comment:
nitpick: the comment is a bit misleading. Because of the way the Serializer from django-filer-rest is working, we have the convention that the self.name has a value only if the file is renamed, otherwise, for any new file upload, we use self.original_name. I think that what you are trying to do here, is that if we replace a file that was renamed before, we need to `erase` the modified name.

Reply:
That is correct, but the comment does not specify why we are doing that only that it overwrites the field. Even if we don't rename the file we still should delete the old metadata of it.

Reply:
Not necessarily. Only because we have that convention with `original_name`. Otherwise, self.name could have the value of the new name set after the upload, and that is that. I mean, essentially, replacing a file could also be seen as a rename, because the id is the same, but the file has a new name.